Adb_usb converter ARM/rp2040 support

ImNoahDev opened the pull request:
This PR adds support for the RP2040 to the ADB-to-USB keyboard converter. The core ADB protocol implimentation has been refactored to be platform-agnostic and will work across all AVR and ARM/ChibiOS platforms.

## Description

- RP2040 support added to ADB-to-USB converter keyboard (keyboards/converter/adb_usb/rp2040/)
- ADB protocol refactored to be platform agnostic 
- Added wiring diagrams and README for RP2040 vased converter. 

All changes have been tested to confirm it builds sucessfully and an adapter has been built and tested using an RP2040-zero board.

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

- Adds ADB keyboard support on RP2040 microcontrollers in the adb_usb converter
- Makes ADB converter code portable to future ARM platforms by refactoring the ADB protocol implimentation to be platform agnostic.

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

ImNoahDev committed:
Refactor ADB USB converter for RP2040 support

- Introduced platform-agnostic GPIO handling for ADB_DATA_PIN in adb.c, allowing compatibility with RP2040.
- Updated adb.h to check for either AVR-style or platform-agnostic pin definitions.
- Added new configuration files for RP2040, including keyboard.json, mcuconf.h, halconf.h, and readme.md detailing hardware requirements and wiring.
- Replaced AVR-specific delay functions with platform-agnostic wait functions.
- Cleaned up includes in various files to remove unnecessary AVR dependencies.

This update enhances the ADB USB converter's flexibility and usability across different platforms.

ImNoahDev committed:
Update readme.md for RP2040 ADB USB converter

- Clarified pin configuration for ADB Data connection.
- Removed optional level shifter section to streamline documentation.
- Emphasized the necessity of a pull-up resistor for reliable operation.

These changes enhance the clarity and usability of the documentation for users setting up the ADB USB converter with RP2040.

ImNoahDev committed:
Update copyright information for Noah Patel in ADB USB converter files

- Corrected the email address for Noah Patel in multiple files to reflect the updated contact: <ImNoahDev@imnoah.com>.
- Ensured consistency across all relevant source files, including adb.c, adb.h, config.h, led.c, matrix.c, and RP2040-specific headers.

This change maintains accurate attribution for contributions to the ADB USB converter project.

ImNoahDev committed:
Add copyright notice to keymap.c for ADB USB converter

- Included the GNU General Public License information and copyright attribution for Jun Wako.
- This update ensures proper licensing and attribution for the keymap.c file in the ADB USB converter project.

ImNoahDev committed:
Enhance ADB USB converter documentation for RP2040 support

- Added detailed instructions for using the ADB USB converter with RP2040 boards, including wiring and safety notes.
- Clarified build commands for both AVR and RP2040 versions.
- Updated the readme to improve usability and provide comprehensive guidance for users.

These changes ensure that users have clear and accurate information for setting up the ADB USB converter across different platforms.

drashna commented on keyboards/converter/adb_usb/rp2040/config.h:
```diff
@@ -0,0 +1,29 @@
+
+/* RP2040 double-tap reset bootloader configuration */
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET
+#define RP2040_BOOTLOADER_DOUBLE_TAP_RESET_TIMEOUT 200U
```
```suggestion
```

The timeout isn't needed. If the double tab reset is defined, then it defaults to 200U

drashna commented on keyboards/converter/adb_usb/rp2040/mcuconf.h:
Also should be removed.

ImNoahDev committed:
Update keyboards/converter/adb_usb/rp2040/config.h

Co-authored-by: Drashna Jaelre <drashna@live.com>

ImNoahDev committed:
Update keyboards/converter/adb_usb/keymaps/default/keymap.c

Co-authored-by: Drashna Jaelre <drashna@live.com>

ImNoahDev committed:
Remove unused RP2040 configuration files halconf.h and mcuconf.h from ADB USB converter directory. These files are no longer needed and their removal helps streamline the project structure.

ImNoahDev committed:
Refactor ADB USB converter to use platform-agnostic wait function

- Replaced AVR-specific delay functions with QMK's wait_us across adb.c for improved compatibility.
- Updated timing calls in various functions to utilize the new wait_us function, enhancing code consistency and maintainability.

These changes streamline the ADB USB converter's implementation, making it more adaptable for different platforms.

ImNoahDev committed:
Update keyboards/converter/adb_usb/readme.md

Co-authored-by: Drashna Jaelre <drashna@live.com>

drashna reviewed (commented):
These changes should allow you to use custom matrix lite, rather than the custom matrix.

drashna commented:
This has a merge conflict that needs resolving.
